Source/VCGeneration/Prune/DataflowAnalysis.cs

```diff
@@ -0,0 +1,61 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace VCGeneration.Prune;
+
+/// <summary>
+/// Allows defining dataflow analysis
+/// </summary>
+abstract class DataflowAnalysis<TNode, TState> {
```

Comment:
This is simple enough that it might not provide a benefit, but have you looked at whether this can be a special case of what's defined in the `AbstractInterpretation` project? Dataflow analysis is generally a special case of abstract interpretation.

Reply:
I had a look at that package but it's a but hard to see how to extract what I need from there. Also, there isn't much code to deduplicate, since this `DataflowAnalysis` is tiny and all it really needs that could already be defined somewhere else, is a control flow graph with `Cmd` as a `Node`.